.github/workflows/rebuild_dictionary.yml

```diff
@@ -0,0 +1,37 @@
+name: Rebuild dictionary
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron:  '0 */12 * * *'
+
+jobs:
+  run:
+    name: Build dictionary
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.head_ref }}
+
+    - name: Set up ruby
+      uses: ruby/setup-ruby@v1
+
+    - name: Install dependencies
+      run: |
+        gem install bundler
+        bundle install --jobs 4 --retry 3
+
+    - name: Rebuild dictionary
+      run: ruby lib/build_local_glossary.rb
+      env:
+        SPREADSHEET_ID: ${{ secrets.SPREADSHEET_ID }}
+        GOOGLE_CLIENT_SECRET: ${{ secrets.GOOGLE_CLIENT_SECRET }}
+
+    - name: Commit the updated dictionary
```

Comment:
I doubt this will work perfectly on the first try, so if reviewers are aware of any gotchas or can spot something brittle in the setup please speak up.

Reply:
I agree that these things never tend to work first time no matter how hard you squint at it. 

I'd suggest using something like [Act](https://github.com/nektos/act) or making a scratch repo to test this and share your results so we can build confidence before we merge it in?